# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?


In [1]:
import pandas as pd
import numpy as np
data=pd.read_csv(r"C:\Users\ALIYA\OneDrive\Documents\Coding_Temple\Week 3\Project\AB_NYC_2019.csv")

In [3]:
display(data.sort_values("calculated_host_listings_count", ascending=False).drop_duplicates("host_name").head(10))

#The bisiest places are all in Manhattan and seem to be just host doing rents for apts

#Which jpsts are the busiest and why
ab_avg=data.groupby(['host_id','host_name'],as_index=False).sum(True).sort_values('reviews_per_month',ascending=False)

#% avaliablity not vacant would be a best metric 

ab_avg[['host_id','host_name','reviews_per_month']].head(5)

#Calculated host listings is total listings made by specific hosts, we did not feel

id                                               name    host_id  \
39773  30937596         Sonder | 11th Ave | Contemporary 1BR + Gym  219517861   
38701  30388079  Spacious UWS 1BR w/ Gym, Walk to Central Park ...  107434423   
13039   9880827    NYC luxury 1 BR apt near the Chrysler Building!   30283594   
42840  33260042     Nice Manhattan room close to Apollo & transpo!  137358866   
33464  26450823       Sonder | Hanover Square | Airy 1BR + Rooftop   12243051   
41896  32564618    Furnished apartment United nation! Doorman 5243   16098958   
28582  22147647                 ULTRA CHIC 2 BEDROOM GRAND CENTRAL   61391963   
16433  13188839      Park Avenue, Luxury 1bd apartment Murray Hill   22541573   
34296  27214615            4BR 4 BATH IN THE HEART OF WEST VILLAGE  200380610   
4516    3121851                Amazing 1BD! Renovated Col Circle!!    1475015   

               host_name neighbourhood_group       neighbourhood  latitude  \
39773       Sonder (NYC)           Manhattan      Hell's Kitchen  40.76037   
38701         Blueground           Manhattan     Upper West Side  40.79040   
13039               Kara           Manhattan             Midtown  40.76693   
42840             Kazuya           Manhattan              Harlem  40.81212   
33464             Sonder           Manhattan  Financial District  40.70506   
41896     Jeremy & Laura           Manhattan             Midtown  40.75619   
28582  Corporate Housing           Manhattan         Murray Hill  40.74933   
16433                Ken           Manhattan         Murray Hill  40.74743   
34296            Pranjal           Manhattan        West Village  40.73495   
4516                Mike           Manhattan      Hell's Kitchen  40.76745   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
39773  -73.99744  Entire home/apt    185              29                  1   
38701  -73.97555  Entire home/apt    320              30                  0   
13039  -73.98139  Entire home/apt    239              30                  0   
42840  -73.94341     Private room     37              30                  0   
33464  -74.00790  Entire home/apt    185              29                  1   
41896  -73.96540  Entire home/apt    135              30                  0   
28582  -73.97670  Entire home/apt    200              30                  0   
16433  -73.97945  Entire home/apt    245              30                  0   
34296  -74.00033  Entire home/apt    483              30                  0   
4516   -73.98489  Entire home/apt     87              30                  8   

      last_review  reviews_per_month  calculated_host_listings_count  \
39773  2019-06-10               1.00                             327   
38701         NaN                NaN                             232   
13039         NaN                NaN                             121   
42840         NaN                NaN                             103   
33464  2018-10-01               0.11                              96   
41896         NaN                NaN                              96   
28582         NaN                NaN                              91   
16433         NaN                NaN                              87   
34296         NaN                NaN                              65   
4516   2018-08-15               0.20                              52   

       availability_365  
39773               332  
38701               293  
13039               352  
42840               223  
33464               220  
41896               306  
28582               310  
16433               365  
34296               364  
4516                311

host_id     host_name  reviews_per_month
34629  219517861  Sonder (NYC)             397.56
35807  244361589       Row NYC             111.72
35389  232251881      Lakshmee              80.63
18626   26432133      Danielle              68.02
31055  137274917         David              62.89

In [18]:
# How many neighborhood groups are available and which shows up the most?

data_neighborhood=data.groupby(['neighbourhood_group','host_id'],as_index=False)
data_neighborhood
data.describe(exclude='number')
data.groupby('neighbourhood_group',as_index=False).count().sort_values('id', ascending=False)[['neighbourhood_group','id']]
#data.groupby('neighbourhood_group',as_index=False).sum().sort_values('reviews_per_month', ascending=False)[['neighbourhood_group','calculated_host_listings_count']]



neighbourhood_group     id
2           Manhattan  21661
1            Brooklyn  20104
3              Queens   5666
0               Bronx   1091
4       Staten Island    373

In [34]:
# Are private rooms the most popular in manhattan?
data = data[["neighbourhood_group", "room_type",'reviews_per_month']].groupby(["neighbourhood_group", 'room_type'],as_index=False).sum(True).sort_values('reviews_per_month',ascending =False)

data[data["neighbourhood_group"] == "Manhattan"]
#Entire home/apt is the most populat

neighbourhood_group        room_type  reviews_per_month
6           Manhattan  Entire home/apt           11378.43
7           Manhattan     Private room            9141.42
8           Manhattan      Shared room             638.23

In [36]:
# Which hosts are the busiest and based on their reviews?
host_reve_avg=data.groupby(['host_id','host_name'],as_index=False).sum(True).sort_values('number_of_reviews',ascending=False)
host_reve_avg[['host_id','host_name','number_of_reviews']].head(5)





host_id                       host_name  number_of_reviews
21304  37312959                            Maya               2273
1052     344035  Brooklyn&   Breakfast    -Len-               2205
18626  26432133                        Danielle               2017
20872  35524316                    Yasu & Akiko               1971
21921  40176101                           Brady               1818

In [43]:
#Which neighorhood group has the highest average price?

data.groupby('neighbourhood_group',as_index=False).mean(True).sort_values('price', ascending=False)[['neighbourhood_group','price']].round(2)


neighbourhood_group   price
2           Manhattan  196.88
1            Brooklyn  124.38
4       Staten Island  114.81
3              Queens   99.52
0               Bronx   87.50

In [46]:
# Which neighbor hood group has the highest total price?
data.groupby('neighbourhood_group',as_index=False).sum(True).sort_values('price', ascending=False)[['neighbourhood_group','price']].round(2)


neighbourhood_group    price
2           Manhattan  4264527
1            Brooklyn  2500600
3              Queens   563867
0               Bronx    95459
4       Staten Island    42825

In [50]:
#Which top 5 hosts have the highest total price?
data.groupby(['host_id',"host_name"],as_index=False).sum(True).sort_values('price',ascending=False)[['host_id', 'host_name','price']].head()


host_id     host_name  price
34629  219517861  Sonder (NYC)  82795
29393  107434423    Blueground  70331
32054  156158778         Sally  37097
34034  205031545    Red Awning  35294
19564   30283594          Kara  33581

In [61]:
# Who currently has no (zero) availability with a review count of 100 or more?
zero_aval=data.groupby(['id','number_of_reviews','availability_365'],as_index=False).count().sort_values('availability_365')[['id','number_of_reviews',"availability_365"]]
mask2=(zero_aval["number_of_reviews"]>=100)&(zero_aval['availability_365']==0)
zero_aval[mask2]






id  number_of_reviews  availability_365
14141  10855464                119                 0
14437  11325107                125                 0
13320  10017359                112                 0
13683  10253315                188                 0
13806  10407723                131                 0
...         ...                ...               ...
6898    4932139                163                 0
6888    4925517                318                 0
7117    5162580                205                 0
6602    4774190                103                 0
7714    5838858                108                 0

[162 rows x 3 columns]

In [65]:
# What host has the highest total of prices and where are they located?
data.groupby(['host_id',"host_name",'neighbourhood_group'],as_index=False).sum(True).sort_values('price',ascending=False)[['host_id', 'host_name','price','neighbourhood_group']].head()



host_id     host_name  price neighbourhood_group
34740  219517861  Sonder (NYC)  82795           Manhattan
29481  107434423    Blueground  69741           Manhattan
34141  205031545    Red Awning  35294           Manhattan
19626   30283594          Kara  33581           Manhattan
32152  156158778         Sally  29194           Manhattan

In [92]:
# When did Danielle from Queens last receive a review?

dan_data = data[(data['host_name']== "Danielle")&(data['neighbourhood_group']=='Queens')&(data['last_review'].isnull() == False)].sort_values("last_review",ascending=False)
dan_data= dan_data.drop_duplicates("host_id")
dan_data[['host_id','host_name','neighbourhood_group','last_review']]

host_id host_name neighbourhood_group last_review
22469   26432133  Danielle              Queens  2019-07-08
33861  201647469  Danielle              Queens  2019-06-20
27021  154256662  Danielle              Queens  2018-01-02

## Further Questions

1. Which host has the most listings?

In [66]:
data.groupby(['host_id',"host_name"],as_index=False).sum(True).sort_values('calculated_host_listings_count',ascending=False)[['host_id', 'host_name','calculated_host_listings_count']].head()


host_id       host_name  calculated_host_listings_count
34629  219517861    Sonder (NYC)                          106929
29393  107434423      Blueground                           53824
19564   30283594            Kara                           14641
31064  137358866          Kazuya                           10609
14429   16098958  Jeremy & Laura                            9216

2. How many listings have completely open availability?

In [94]:
zero_aval=data.groupby(['id','calculated_host_listings_count','availability_365'],as_index=False).count().sort_values('availability_365')[['id','calculated_host_listings_count',"availability_365"]]
mask2=(zero_aval['availability_365']==365)
zero_aval[mask2].count()


id                                1295
calculated_host_listings_count    1295
availability_365                  1295
dtype: int64

3. What room_types have the highest review numbers?

In [69]:
data[["neighbourhood_group", "room_type",'reviews_per_month']].groupby([ 'room_type'],as_index=False).sum(True).sort_values('reviews_per_month',ascending =False)


room_type  reviews_per_month
0  Entire home/apt           26565.34
1     Private room           25529.62
2      Shared room            1245.08

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please discribe them here.

-- Add your conclusion --
Question 1:host:219517861

Question 2:	neighbourhood_group	id
        2	Manhattan	21661
        1	Brooklyn	20104
        3	Queens	5666
        0	Bronx	1091
        4	Staten Island	373
Question 3: Entire home/apt is the most popular

Question 4: 37312959	Maya

Question 5:	Manhattan

Question 6:Manhattan 4264527

Question 7:

	    host_id	host_name	price

34629	219517861	Sonder (NYC)	82795

29393	107434423	Blueground	70331

32054	156158778	Sally	37097

34034	205031545	Red Awning	35294

19564	30283594	Kara	33581


Question 8:161 listings

Question 9: 219517861	Sonder (NYC)	82795	Manhattan

Question 10:
        host_id	host_name	neighbourhood_group	last_review

22469	26432133	Danielle	Queens	2019-07-08

33861	201647469	Danielle	Queens	2019-06-20

27021	154256662	Danielle	Queens	2018-01-02



Futher Questions

Question 1: 

host_id	host_name	calculated_host_listings_count

219517861	Sonder (NYC)	106929

Question 2: 1295
Question 3: Entire home/apt	

